Update README (#102)
kbrew-dev/kbrew: README.md

> 

Reviewer: ```suggestion
```

# kbrew

[![CI](https://github.com/kbrew-dev/kbrew/actions/workflows/go.yml/badge.svg)](https://github.com/kbrew-dev/kbrew/actions/workflows/go.yml) [![Go Report Card](https://goreportcard.com/badge/github.com/kbrew-dev/kbrew)](https://goreportcard.com/report/github.com/kbrew-dev/kbrew)
[![Release Version](https://img.shields.io/github/v/release/kbrew-dev/kbrew?label=kbrew)](https://github.com/kbrew-dev/kbrew/releases/latest)
[![License](https://img.shields.io/github/license/kbrew-dev/kbrew?color=light%20green&logo=github)](https://github.com/kbrew-dev/kbrew/blob/main/LICENSE)

![kbrew-logo](./images/kbrew-logo.png)

> kbrew is a Kubernetes tool that aims to make installing any complex stack in any cloud possible with `one step` (And yes we are definitely inspired by Homebrew from MacOS)

Reviewer: ```suggestion
kbrew is a CLI tool for Kubernetes which makes installing any complex stack easy in `one step` (And yes we are definitely inspired by Homebrew from MacOS)
```

kbrew is to Kubernetes what Homebrew is to MacOS - a simple and easy to use package manager which hides the underlying complexity.

> Let's talk in the context of an example of installing Kafka on a Kubernetes cluster

Author: ```suggestion
Let's take the example of installing a Kafka on a Kubernetes cluster. If you are a developer trying this on a non prod environment you want a quick and simple way to set it up. But a typical process looks like this:
```

```
## Helm chart or operator or Manifests - all abstracted

> kbrew abstracts the underlying chart or operator or manifest and gives you a recipe to install a stack with all basic configurations done.

Author: ```suggestion
kbrew abstracts the underlying chart or operator or manifest and gives you a recipe to install a stack with all specific configurations done. You can trust that the recipe `just works`!
```

  install     Install application
  remove      Remove application
  search      Search application
  update      Update kbrew and recipe registries
  version     Print version information

Flags:
  -c, --config string       config file (default is $HOME/.kbrew.yaml)
      --config-dir string   config dir (default is $HOME/.kbrew)
      --debug               enable debug logs
  -h, --help                help for kbrew
  -n, --namespace string    namespace

Use "kbrew [command] --help" for more information about a command.
```

For a quick demo, watch: https://www.youtube.com/watch?v=pWRZhZgfYSw 

### Commonly used commands

#### kbrew search

Searches for a recipe for the given application. Lists all the available recipes if no application name is passed.

#### kbrew info

Prints applications details including registry and dependency information. 

#### kbrew install

Installs a recipe in your cluster with all pre & posts steps and applications.

#### kbrew update

Checks for kbrew updates and upgrades automatically if a newer version is available.
Fetches updates for all the kbrew recipe registries

#### kbrew remove 

Uninstalls the application and its dependencies.

> ## Workflow

Author: This should be called `Working of a Kbrew Recipe` or `Genesis of a Kbrew Recipe` 
Workflow is very overloaded and does not convey clearly what we want to say IMHO.

## CLI Usage

```
$ kbrew --help

> TODO: Long description

Reviewer: ```suggestion
A Kubernetes tool that aims to make installing any complex stack in any cloud possible with one step.
```
This is getting updated with - https://github.com/kbrew-dev/kbrew/pull/109
